packages/grafana-ui/src/components/Logs/LogRowMessage.tsx

```diff
@@ -70,11 +80,13 @@ class UnThemedLogRowMessage extends PureComponent<Props> {
       contextIsOpen,
       showContextToggle,
       wrapLogMessage,
+      prettifyLogMessage,
       onToggleContext,
     } = this.props;
 
     const style = getLogRowStyles(theme, row.logLevel);
     const { entry, hasAnsi, raw } = row;
+    const jsonEntry = prettifyLogMessage && isJson(entry) ? JSON.stringify(JSON.parse(entry), undefined, 2) : entry;
```

Comment:
not a biggie and i'm not sure about performance implications of parsing multiple times the same string, but we are parsing `entry` twice in the worst case scenario.

can we reduce it to one by doing, for example:
```ts
const jsonEntry = prettifyLogMessage ? prettify(entry) : entry;
```
where `prettify` is something like:
```ts
function prettify(str: string) {
  try {
    return JSON.stringify(JSON.parse(str), undefined, 2)
  } catch {
    return str;
  }
}
```
?

Reply:
Yeah you're right, i should change that

Reply:
There is already such a function :) https://github.com/grafana/grafana/blob/abb1c107767e8011b44078f1212ce59cf718374c/public/app/core/utils/explore.ts#L195